src/metpy/plots/declarative.py

```diff
@@ -1332,30 +1419,40 @@ def griddata(self):
     def plotdata(self):
         """Return the data for plotting.
 
-        The data array, x coordinates, and y coordinates.
+        The dimension coordinates and data arrays.
 
         """
-        x = self.griddata[0].metpy.x
-        y = self.griddata[0].metpy.y
-
-        if self.earth_relative:
-            x, y, _ = ccrs.PlateCarree().transform_points(self.griddata[0].metpy.cartopy_crs,
-                                                          *np.meshgrid(x, y)).T
-            x = x.T
-            y = y.T
-        else:
-            if 'degree' in x.units:
-                x, y, _ = self.griddata[0].metpy.cartopy_crs.transform_points(
-                    ccrs.PlateCarree(), *np.meshgrid(x, y)).T
-                x = x.T
-                y = y.T
-
-        if x.ndim == 1:
-            xx, yy = np.meshgrid(x, y)
-        else:
-            xx, yy = x, y
+        x_like = self.griddata[0][self.griddata[0].dims[1]]
+        y_like = self.griddata[0][self.griddata[0].dims[0]]
+
+        if (
+            x_like.name == self.griddata[0].metpy.find_axis_name('x')
+            and y_like.name == self.griddata[0].metpy.find_axis_name('y')
+        ):
```

Comment:
This doesn't seem right. `x_like`/`y_like` is relying on the order, while we then look at finding them by name, but then assume those are the same.

Reply:
I think this is currently a future proofing set up to be able to plot wind barbs not just on XY planes, but time-height cross sections or regular cross sections. We don't currently have that ability, so that is why the if statement checking for matching of x and y coordinates doesn't have an `elif` statement at this time. @jthielen might be able to provide more clarity here.

Reply:
Yes, this was an attempt at future-proofing for non-map 2D plots. As described in the comment below, this is a conditional check to only carry out the earth/grid relative adjustments if we are in the plane of the plot (i.e., a map rather than a general 2D plot).

However, the confusion is definitely warranted, since this doesn't match the intent...or at the very least, won't work for the desired generalized case. Since `find_axis_name` only finds dimension coordinate names, as-is this will error out when not in an yx case, since then y and/or x are not there.

The goal is to check that we are in yx space (or technically also xy, but my initial attempt neglected the non-standard dimension order), which gets a bit tricky since we have to use `find_axis_name` or `find_axis_number` which enforce dimension coordinates rather than MetPy's coordinate helpers for `y` and `x` (which just enforce 0D or 1D coords). So, this may need to change to something like the following

```python
check_earth_relative = False
try:
    plot_x_dim = self.griddata[0].metpy.find_axis_number('x')
    plot_y_dim = self.griddata[0].metpy.find_axis_number('y')
    check_earth_relative = True
except ValueError:
   plot_x_dim = 1
   plot_y_dim = 0

x_like = self.griddata[0][self.griddata[0].dims[plot_x_dim]]
y_like = self.griddata[0][self.griddata[0].dims[plot_y_dim]]

if check_earth_relative:
   ...
```